tests/data/snake_oil_data.py

```diff
@@ -12,6 +12,24 @@ def content(self):
         return self._content
 
 
+ensemble_for_axis_labels_tests = "default3"
+axis_labels_test_index_response = "FGPT"
+axis_labels_test_date_response = "FOPR"
```

Comment:
I would just use the strings and not add these variables. Adding them dose not improve code readability.

For example, now you write:

```
all_ensemble_names = [
    "default",
    "default_smoother_update",
    ensemble_for_axis_labels_tests,
    "nr_42",
]
```
and the above list contains 3 strings and the variable `ensemble_for_axis_labels_tests` this does not seem very consistent.

Reply:
Hm, yes, I agree - but I wasn't going for readability yet. This is an attempt to move towards data driven tests. 
I understand it makes for incohesive style in the code right now, and I'm hoping that this can be resolved with a refactoring PR as hinted at above.
also, it's mostly DRYness within the same file I'm thinking of with lines 15-17.
furthermore - this (the variable `all_ensemble_names`) actually serves to reduce brittleness. one unrelated test broke, that chose certain ensembles, and expected to just find another ensemble left, all hard coded.

please let me know if i was successful with defending this position.

Reply:
right, I too think hard coding them in the test is OK - I just don't think there's a need to do it many times, once should be enough.

I don't know, maybe this is a bit philosophical, I see the point of loading the ensemble names like you suggest. When hard-coding the list, I think I had in mind what you mention with having it explicit in the code, so that you notice when something breaks.

And I like hyper-explicit names : D it is strange to use that name in another test, and I would solve that problem by doing something like

```
ensemble_for_that_other_kind_of_test = ensemble_for_axis_labels_test
```

(which gives you a clear dependency, which one might wonder is that something you want, ...)

OK so let me summarize a bit:

- I don't want to use the literal string more than once in the test file -> put it in variable
- I don't mind much about the name of the variable, so I can change that, if desired
- I still think that it's reasonable to use that list of ensembles in the example where I used it, because if you leave it as it was, by changing the test data like I did, you break that test in a "formal" way, you're not breaking anything meaningful or functional.
- How we define the list of ensembles, I don't really mind I suppose, we can do it dynamically via that API call, or have it hard-coded in a list...

what do you think? can i leave the variables? should i change the names and / or the definition of the list? or should I scrape it (the variables) all, and leave it as literal strings everywhere, and the literal list of ensembles in that one test?

Reply:
not sure if you wanted to add anything here, @DanSava, so I'm gonna leave it open for now

Reply:
> what do you think? can i leave the variables? should i change the names and / or the definition of the list? or should I scrape it (the variables) all, and leave it as literal strings everywhere, and the literal list of ensembles in that one test?

I think you should do what makes you happy what makes the code logical and easy to understand what makes the code readable and achieve this by removing logic from the codebase not adding it, in this order. 

I guess what I am saying is do what makes you happy :smiley: 

 And if you like hyper-explicit names I would suggest using something similar in each test:

```
ensemble_for_axis_labels_test = ensemble_general_constant_varaible_name_defined_somewhere_in_the_conftest_file
```

Reply:
ok thank you : )

Reply:
gonna resolve this